gslib/tests/test_rsync.py

```diff
@@ -1460,6 +1460,7 @@ def test_bucket_to_dir_preserve_posix_no_errors(self):
     """Tests that rsync -P works properly with default file attributes."""
     bucket_uri = self.CreateBucket()
     tmpdir = self.CreateTempDir()
+    PRIMARY_GID = os.stat(tmpdir).st_gid
```

Comment:
If we're overwriting `PRIMARY_GID` here, there doesn't seem to be a point in keeping it (or the method it calls, `GetPrimaryGid()`) in `gslib/tests/util.py`.  To that effect, the naming convention for this variable can be changed, from `PRIMARY_GID` to `primary_gid`.

Similarly, it looks like we only use `NON_PRIMARY_GID` in these two tests.  Thus, I'd suggest keeping the `GetNonPrimaryGid()` method in `gslib/tests/util.py`, getting rid of the `NON_PRIMARY_GID` module member there (since it's just a LazyWrapper that isn't all that useful), and just using an expression like this at the top of these two tests: 

```
non_primary_gid = GetNonPrimaryGid()
```

Reply:
Got it! Will do :+1:

Reply:
Staged changes on a test branch, running tests: gpaste/6197306502152192

Reply:
A couple more tests are failing post-refactor surrounding tracker files, going to smooth that out and test it again

Reply:
It was a silly typo I think, here's a new batch of tests targeting the `GCS/gsutil : macGidFix` branch: https://paste.googleplex.com/5510027513167872

Reply:
Those runs seemed to go much better. :) Mac and Linux both were all green (except a single flaked io.close on linux that we're aware can happen) and Windows green except resumable test failure (which we have a bug for and will work on next) :+1: 

Will merge the changes on the testing branch into this PR shortly.

Reply:
Commit 3bbdad74bce8fd6730e325f5228da4172f529609 has the refactor and is now in this PR :muscle:

Reply:
Oh wait, this refactor seems to have broken test_tab_complete and test_stat somehow. Investigating.

Reply:
I'm kind of lost how these changes would have impacted either of the above tests. I'm going to ask Kokoro to rerun tests on this to make sure it wasn't a weird test flake.

Reply:
Rerunning tests and the errors still reappear. Specifically, all Mac XML tests encounter errors on `test_tabcomplete.test_subcommands` and all Linux XML tests encounter errors on `test_stat_bucket_wildcard`.

Test logs: https://paste.googleplex.com/4826875685765120

Reply:
Oh! I re-ran against master, and the Tabcomplete and stat_bucket_wildcard issues are there too. So not related to PR which makes me feel less like I've gone nuts. :-)  I'll open bugs for those to smooth out.